### PR TITLE
Convert Gradle script plugin for generating UniFfi bindings into a composite build.

### PR DESCRIPTION
**Pull request by kirillzh**
### Description

`buildSrc` has a [major drawback](https://proandroiddev.com/stop-using-gradle-buildsrc-use-composite-builds-instead-3c38ac7a2ab3) of invalidating entire project's cache after a chance in the `buildSrc` itself. This is a probably not a big issue for the project of this size, but a way to fix this is to implement plugin as a standalone, composite build, which also happens to be a more idiomatic way of [implementing plugins](https://docs.gradle.org/current/userguide/custom_plugins.html#sec:custom_plugins_standalone_project).

### Notes to the reviewers

These are the rough steps that I took for converting the plugin:
1. . Create new standalone `plugins` Gradle module for the plugin
2. Create two new `UniFfiJvmPlugin` and `UniFfiAndroidPlugin` Gradle plugin classes, copy-and-pasting existing script plugin implementations
3. Register two new plugins using the same IDs as convention plugins: `org.bitcoindevkit.plugins.generate-jvm-bindings` and `org.bitcoindevkit.plugins.generate-android-bindings`
4. Delete `buildSrc`

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
Very neat. Will review later today but I think this is a great step forward.

**Review by thunderbiscuit — APPROVED** [review]
This is awesome. 🔥 I really enjoy digging into Gradle, and I had not really looked into how to make our plugins into their own standalone projects. Thanks for showing us how it's done!

My only comment slash small nit is that I think we should keep the style consistent and apply 4 spaces for the indents in the `build.gradle.kts` and the `settings.gradle.kts` files. Otherwise this is good to go.

**Review by notmandatory — APPROVED** [review]
utACK cc20cf6b7f9421fb282199bde8826c0809167ba1

Looks good!

**Comment by thunderbiscuit** [comment]
@kirillzh I can't seem to push the small fix for the indentation to your branch. Do you mind adding it yourself? I'll merge this right after.

It's basically just using 4 spaces indent for the `build.gradle.kts` and the `settings.gradle.kts` files.

**Comment by kirillzh** [comment]
@thunderbiscuit, reformatted `plugins/build.gradle.kts` and `plugins/settings.gradle.kts` 👍🏻

**Comment by thunderbiscuit** [comment]
The CI is failing with this error:
```sh
   Compiling bdk-ffi-bindgen v0.2.0 (/home/runner/work/bdk-kotlin/bdk-kotlin/bdk-ffi/bdk-ffi-bindgen)
error: linker `aarch64-linux-android21-clang` not found
  |
  = note: No such file or directory (os error 2)

error: could not compile `bdk-ffi-bindgen` due to previous error
warning: build failed, waiting for other jobs to finish...

> Task :android:buildAndroidAarch64Binary FAILED
error: could not compile `bdk-ffi` due to previous error
```

I will investigate tomorrow.

**Comment by thunderbiscuit** [comment]
I managed to fix the CI. Will open the PR tomorrow, and then we can finally merge this thing!

**Comment by notmandatory** [comment]
@kirillzh the CI issue was fixed in #77 so ready for you to rebase this one, thanks!

**Comment by thunderbiscuit** [comment]
This time around the error is
```shell
* Where:
Build file '/home/runner/work/bdk-kotlin/bdk-kotlin/plugins/build.gradle.kts' line: 1

* What went wrong:
Plugin [id: 'org.gradle.kotlin.kotlin-dsl', version: '2.1.7'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Included Builds (None of the included builds contain this plugin)
- Plugin Repositories (could not resolve plugin artifact 'org.gradle.kotlin.kotlin-dsl:org.gradle.kotlin.kotlin-dsl.gradle.plugin:2.1.7')
  Searched in the following repositories:
    Gradle Central Plugin Repository
```

**Comment by thunderbiscuit** [comment]
Not sure where the decision to pull this particular version of the plugin is made... do you mind trying to set the version explicitly?
```gradle
    `kotlin-dsl` version "2.4.1"
    // id("org.gradle.kotlin.kotlin-dsl") version "2.4.1"
```

**Comment by kirillzh** [comment]
> This time around the error is
> 
> ```shell
> * Where:
> Build file '/home/runner/work/bdk-kotlin/bdk-kotlin/plugins/build.gradle.kts' line: 1
> 
> * What went wrong:
> Plugin [id: 'org.gradle.kotlin.kotlin-dsl', version: '2.1.7'] was not found in any of the following sources:
> 
> - Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
> - Included Builds (None of the included builds contain this plugin)
> - Plugin Repositories (could not resolve plugin artifact 'org.gradle.kotlin.kotlin-dsl:org.gradle.kotlin.kotlin-dsl.gradle.plugin:2.1.7')
>   Searched in the following repositories:
>     Gradle Central Plugin Repository
> ```

Hm, interestingly I am not seeing this error locally.

File: plugins/settings.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+dependencyResolutionManagement {
+    repositories {
+        mavenCentral()
+        google()
+        gradlePluginPortal()
```
Comment (kirillzh) [inline plugins/settings.gradle.kts:5]:
```
* What went wrong:
Plugin [id: 'org.gradle.kotlin.kotlin-dsl', version: '2.1.7'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Included Builds (None of the included builds contain this plugin)
- Plugin Repositories (could not resolve plugin artifact 'org.gradle.kotlin.kotlin-dsl:org.gradle.kotlin.kotlin-dsl.gradle.plugin:2.1.7')
  Searched in the following repositories:
    Gradle Central Plugin Repository
```

@thunderbiscuit, adding `gradlePluginPortal()` repository here should fix the problem, though not entirely sure why 😅 

Usually, `plugins` adds Gradle plugin repository implicitly so adding this explicitly should not be necessary 🤔 🤷🏻‍♂️ 
> By default, the plugins {} DSL resolves plugins from the public [Gradle Plugin Portal.](https://plugins.gradle.org/)

Comment (kirillzh) [reply plugins/settings.gradle.kts]: Actually, looks like Gradle plugin repository was down yesterday, so `gradlePluginPortal()` is in fact not necessary here, removing `gradlePluginPortal()`!

**Review by notmandatory — APPROVED** [review]
ReAck 989b7339a0e7e44ec56f3d3056180bc1ce278de8

Looks ready, @thunderbiscuit  or I will merge tomorrow if no objections

**Pull request merged**